x-pack/plugins/data_quality/public/routes/dataset_quality/context.tsx

```diff
@@ -43,6 +43,14 @@ export function DatasetQualityContextProvider({
         initialState,
       });
       datasetQualityController.service.start();
+
+      if (initialState?.flyout?.dataset) {
+        datasetQualityController.service.send({
+          type: 'OPEN_FLYOUT',
+          dataset: initialState?.flyout?.dataset,
```

Comment:
You don't need optional chaining here as you are already checking for its presence